app/routers/weekly_tasks.py

```diff
@@ -0,0 +1,201 @@
+        "Thu": thu,
+        "Fri": fri,
+        "Sat": sat,
+        "Sun": sun
+    }
+    days_list = [day for day, is_checked in days_dict.items() if is_checked]
+    days = ", ".join(days_list)
+    return days
+
+
+@router.get("/")
+def weekly_tasks_manager(
+        request: Request,
+        session=Depends(get_db),
+        demo_user=Depends(get_placeholder_user)):
+
+    user = get_user(demo_user, session)
+
+    # TODO: Move the below function to a compatible location
+    # Need to run regularly whenever there are tasks on the week
+    # Or will run on the background after the user left the
+    # weekly-tasks manager page
+    # function:
+    # generate_tasks(user, session)  # imported from app.internal.weekly_tasks
+    # session.close()
+
+    return templates.TemplateResponse("weekly_tasks_manager.html", {
+        "request": request,
+        "weekly_tasks": user.weekly_tasks
+    })
+
+
+@router.get("/add")
+def weekly_task_add(request: Request):
+
+    checked_days = get_checked_days()
+    return templates.TemplateResponse("add_edit_weekly_task.html", {
+        "request": request,
+        "weekly_task": None,
+        "mode": "add",
+        "checked_days": checked_days
+    })
+
+
+@router.post("/delete")
+def weekly_task_remove(
+        session=Depends(get_db),
+        remove_id: int = Form(...)):
+
+    remove_weekly_task(remove_id, session)
+    session.close()
+    url = router.url_path_for("weekly_tasks_manager")
+    return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+
+
+@router.post("/edit")
+def weekly_task_edit(
+        request: Request,
+        session=Depends(get_db),
+        edit_id: int = Form(...)):
+
+    weekly_task = session.query(WeeklyTask).filter_by(id=edit_id).first()
+    checked_days = get_checked_days(weekly_task.days)
+    return templates.TemplateResponse("add_edit_weekly_task.html", {
+        "request": request,
+        "weekly_task": weekly_task,
+        "mode": "edit",
+        "checked_days": checked_days
+    })
+
+
+@router.post("/make-change")
+def weekly_task_make_change(
+        request: Request,
+        session=Depends(get_db),
+        demo_user=Depends(get_placeholder_user),
+        title: str = Form(None),
+        sun: bool = Form(False),
+        mon: bool = Form(False),
+        tue: bool = Form(False),
+        wed: bool = Form(False),
+        thu: bool = Form(False),
+        fri: bool = Form(False),
+        sat: bool = Form(False),
+        content: str = Form(None),
+        is_important: bool = Form(False),
+        the_time: datetime.time = Form(None),
+        weekly_task_id: int = Form(...),
+        mode: str = Form(...)):
+
+    user = get_user(demo_user, session)
+    days = get_days_string(
+        sun, mon, tue, wed, thu, fri, sat
+    )
+
+    weekly_task = weekly_task_from_input(
+        user,
+        title, days,
+        content, the_time,
+        is_important,
+        weekly_task_id=weekly_task_id
+    )
+
+    made_change = False
```

Comment:
Remove

Reply:
changed the names so it will be more clear, please look again, tell me what you think :)